tensorflow/python/util/util.cc

```diff
@@ -213,9 +213,9 @@ int IsInstanceOfRegisteredType(PyObject* obj, const char* type_name) {
 // Returns 1 if `o` is considered a mapping for the purposes of Flatten().
 // Returns 0 otherwise.
 // Returns -1 if an error occurred.
-int IsMappingHelper(PyObject* o) {
+int IsNestCompatibleMappingHelper(PyObject* o) {
```

Comment:
There are a few other C++ files that use this function, so you probably want to add a new one like this instead.

Then you'll need to update the name at line 122.

Lastly, you'll need to change line 145 so that is creates an new empty object then calls `update` on it, which is part of the `MutableMapping]` interface:

```
new_mapping = instance_type()
new_mapping.update(result)
return new_mapping
```

Reply:
I created a new function to support `MutableMapping` but I couldn't figure out where to make the other two changes you requested.

Reply:
Start by adding it around line 112, `_is_nest_compatible_mapping = _pywrap_utils.IsNestCompatibleMapping`; the tests will tell you when everything is wired properly.

Reply:
I ran the tests and it ran successfully.

Reply:
Getting this error 
```
AttributeError: module 'tensorflow.python._pywrap_utils' has no attribute 'IsNestCompatibleMapping'
```
Where is `_pywrap_utils` defined? I looked and could not find it.

Reply:
It seems to be generated by this target: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/BUILD#L649

Reply:
```error: 'IsNestCompatibleMapping' is not a member of 'tensorflow::swig'``` . How can I add the newly formed method `isNestCompatibleMapping` to the `swig` namespace?

Reply:
See how it was done for e.g. `IsMapping`. I think it's in the same file.